File: tests/07_phases_ompss.dg/fortran/success_task_26.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM P
    INTEGER I,J

    I = 0
    J = 1
    !$OMP TASK INOUT(i,j)
        if (i == 0) J = 0

        !$OMP TASK INOUT(i, j)
        I = I + 1
        !$OMP END TASK
        !$OMP TASKWAIT

       J = J + 1
    !$OMP END TASK
    !$OMP TASKWAIT

    if (I /= 1 .or. J /= 1) STOP 1
END PROGRAM P

File: tests/07_phases_ompss.dg/c/success_task_26.c
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
test_nolink=yes
</testinfo>
*/

// This test was designed to stress our transformation of the final clause.
// There is no need to execute it
int main()
{
    int x = 0;
    #pragma omp task inout(x)
    {
        x++;
        #pragma omp taskwait inout(x)
    }
    #pragma omp taskwait
}
